File: frontend/src/types/course.ts
// ─── Course Category ──────────────────────────────────────────────────────────

export interface CourseCategoryDto {
  id: number;
  name: string;
  slug: string;
  description?: string;
  icon?: string;
  displayOrder: number;
  isActive: boolean;
  courseCount: number;
  createdAt: string;
}

// ─── Course ───────────────────────────────────────────────────────────────────

export interface CourseDto {
  id: number;
  categoryId: number;
  categoryName: string;
  categorySlug: string;
  title: string;
  slug: string;
  description?: string;
  difficulty: string;
  estimatedHours?: number;
  isPublished: boolean;
  displayOrder: number;
  moduleCount: number;
  topicCount: number;
  createdAt: string;
  updatedAt: string;
}

// ─── Course Module ────────────────────────────────────────────────────────────

export interface TopicSummaryDto {
  id: number;
  moduleId: number;
  title: string;
  slug: string;
  difficulty: string;
  estimatedMinutes?: number;
  displayOrder: number;
  isPublished: boolean;
}

export interface CourseModuleDto {
  id: number;
  courseId: number;
  title: string;
  description?: string;
  displayOrder: number;
  topics?: TopicSummaryDto[];
}

export interface CourseDetailDto extends CourseDto {
  modules: CourseModuleDto[];
}

export interface CourseRequest {
  categoryId?: number;
  title: string;
  slug: string;
  description?: string;
  difficulty?: string;
  estimatedHours?: number;
  isPublished?: boolean;
  displayOrder?: number;
}

export interface CourseModuleRequest {
  title: string;
  description?: string;
  displayOrder?: number;
}

File: backend/src/main/java/com/vcube/academy/entity/Course.java
package com.vcube.academy.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "courses")
@EntityListeners(AuditingEntityListener.class)
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private CourseCategory category;

    @Column(name = "title", nullable = false, length = 200)
    private String title;

    @Column(name = "slug", nullable = false, unique = true, length = 200)
    private String slug;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "difficulty", nullable = false, length = 20)
    @Builder.Default
    private String difficulty = "BEGINNER";

    @Column(name = "estimated_hours")
    private Integer estimatedHours;

    @Column(name = "is_published", nullable = false)
    @Builder.Default
    private Boolean isPublished = false;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private Integer displayOrder = 0;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @OrderBy("displayOrder ASC")
    @Builder.Default
    private List<CourseModule> modules = new ArrayList<>();

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private Instant updatedAt;
}
